src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -744,51 +842,68 @@ await ExecuteWithFailOverPolicyAsync(
                 }
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
+
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
-                _watchedSettings = watchedSettings;
+
+                _watchedIndividualKvs = watchedIndividualKvs;
+                _watchedKvEtags = watchedKvEtags;
+                _watchedFfEtags = watchedFfEtags;
                 _mappedData = mappedData;
             }
         }
 
-        private async Task<Dictionary<string, ConfigurationSetting>> LoadSelectedKeyValues(ConfigurationClient client, CancellationToken cancellationToken)
+        private async Task<Dictionary<string, ConfigurationSetting>> LoadSelected(
+            ConfigurationClient client,
+            Dictionary<KeyValueSelector, IEnumerable<MatchConditions>> watchedKvEtags,
+            Dictionary<KeyValueSelector, IEnumerable<MatchConditions>> watchedFfEtags,
+            IEnumerable<KeyValueSelector> selectors,
+            CancellationToken cancellationToken)
         {
-            var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
-
-            // Use default query if there are no key-values specified for use other than the feature flags
-            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => selector.KeyFilter == null ||
-                !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
+            Dictionary<string, ConfigurationSetting> data = new Dictionary<string, ConfigurationSetting>();
 
-            if (useDefaultQuery)
+            foreach (KeyValueSelector loadOption in selectors)
```

Comment:
```suggestion
            foreach (KeyValueSelector selector in selectors)
```

Reply:
we're using selector for `SettingSelector` but I could rename `kvSelector` and `settingSelector`

Reply:
Ah, I see, but it looks like that selector can be created inline right? Then the name wouldn't be occupied.